frame/composable-traits/src/dex.rs

```diff
@@ -111,3 +112,28 @@ pub struct ConstantProductPoolInfo<AccountId, AssetId> {
 	/// Amount of the fee pool charges for the exchange
 	pub fee: Permill,
 }
+
+#[derive(Encode, Decode, TypeInfo, Clone, PartialEq, Eq, RuntimeDebug)]
+pub enum DexRouteNode<PoolId> {
+	Curve(PoolId),
```

Comment:
This would force both DEXes to use the same PoolId, better to make it take two generics

Reply:
what if in future we may have one more AMM? @KaiserKarel

Reply:
yeah, having poolid seems resonable tradeoff. if new pallet to be integrated, than it should be refactoed to be generic on poolid and support same poold id type.  If pallet will use different model of poolid thing, than may add <PoolId, SomeOtherPoolId> later